File: #0214.shortest-palindrome.cpp
class Solution {
public:
    string shortestPalindrome(string s) {

        int size = s.size();
        int index = 0;
        for(int i = size - 1; i >= 0; --i) {
            if (s[index] == s[i]) {
                ++index;
            }
        }

        if (index == size) {
            return s;
        }

        string target = s.substr(index);
        string temp = target;
        reverse(temp.begin(), temp.end());
        return temp + shortestPalindrome(s.substr(0, index)) + target;
    }
};

File: #0214.shortest-palindrome.js
/**
 * @param {string} s
 * @return {string}
 */
var shortestPalindrome = function(s) {
    const reserse = s.split('').reverse().join('');
    const len = s.length;
    let substr = '';
    for (let i = len - 1; i >= 0; i--) {
        const short = i + 1;
        if (s.slice(0, short) === reserse.slice(len - short, len)) {
            substr = reserse.slice(0, len - short);
            break;
        }
    }
    return substr + s;
};
